src/master-playlist-controller.js

```diff
@@ -674,6 +671,28 @@ export class MasterPlaylistController extends videojs.EventTarget {
     this.trigger('sourceopen');
   }
 
+  /**
+   * handle the sourceended event on the MediaSource
+   *
+   * @private
+   */
+  handleSourceEnded_() {
+    if (!this.inbandTextTracks_.metadataTrack_) {
```

Comment:
mostly taken and modified from the [`sourceended` handler in `html-media-source`](https://github.com/videojs/http-streaming/blob/master/src/mse/html-media-source.js#L220-L231)